docs/tutorials/hackage-stackage.md

```diff
@@ -32,3 +32,34 @@ in ...
 This way you can change the revisions of `hackage.nix` and `stackage.nix` are fetched from without changing `haskell.nix`.
 
 However, bear in mind that Stackage refers to Hackage, so your Stackage should never be newer than Hackage.
+
+### Pinning the Hackage version
+
+Sometimes you might want to use a chosen version of Haskell.nix with a
+recent update of Hackage or Stackage. This can be done with JSON pins:
+
+```bash
+nix-prefetch-git https://github.com/input-output-hk/hackage.nix | tee hackage-src.json
+nix-prefetch-git https://github.com/input-output-hk/stackage.nix | tee stackage-src.json
+```
+
+The resulting JSON files will correspond to the latest revision of
+[hackage.nix][] and [stackage.nix][]. See
+[Architecture](../architecture.md) for more information about how
+these Git repositories correspond to the actual Hackage and Stackage.
+
+[hackage.nix]: https://github.com/input-output-hk/hackage.nix
+[stackage.nix]: https://github.com/input-output-hk/stackage.nix
+
+```nix
+{ pkgs ? import <nixpkgs> (haskellNixArgs // { overlays = haskellNixArgs.overlays ++ [
+  (self: super: {
+    haskell-nix = super.haskell-nix // {
+      hackageSourceJSON  = ./hackage-src.json;
+      stackageSourceJSON = ./stackage-src.json;
+    };
+  })]; })
+, haskellNixArgs ? (import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz) {}).nixpkgsArgs
+}:
+  pkgs
```

Comment:
This is different to the way described in the existing docs - any reason for that? (I prefer the current way)

Reply:
This was moved from getting started. I'm a bit confused as there are two ways to specify, via hackageSourceJSON and sourcesOverride. We should probably pick one and use it consistently.

Reply:
Ah, I mised that :) We're gradually moving towards `niv` for everything, so `hackageSourceJSON` will hopefully vanish in favour of `sourceOverrides`.

Maybe we can just point to the other doc about pinning hackage/stackage?

Reply:
What other docs do you have in mind? 

I agree about `sourceOverrides`, it's a more general interface.

Reply:
[hackage-stackage.md](https://github.com/input-output-hk/haskell.nix/pull/702/files#diff-bf5433094c64528476870c9d0e1075ebR1)

Reply:
Wait, this is the same file, it's just a few paragraphs up!